tests/unit/generateCertificateNumber.unitTest.ts

```diff
@@ -52,4 +53,51 @@ describe("TestResultsService calling generateExpiryDate", () => {
         });
     });
 
+    context("when inserting a testResult that is a vehicleType of hgv and it contains at least one Roadworthiness test type and the test result on the Roadworthiness test type is fail", () => {
+        it("then a testNumber is generated and the inserted test result should not have a certificate number", () => {
+            const testFailedResultWithRoadworthinessTestType = cloneDeep(testResultsPostMock[6]);
+            testFailedResultWithRoadworthinessTestType.vehicleType = "hgv";
+            testFailedResultWithRoadworthinessTestType.testTypes[0].testTypeId = "122";
+            testFailedResultWithRoadworthinessTestType.testTypes[0].testNumber = "W01A00209";
+            testFailedResultWithRoadworthinessTestType.testTypes[0].testResult = "fail";
+            testFailedResultWithRoadworthinessTestType.testTypes[0].certificateNumber = null;
+
+            expect.assertions(2);
+            const updatedResult =  testResultsService.generateCertificateNumber(testFailedResultWithRoadworthinessTestType);
+            expect(updatedResult.testTypes[0].testTypeId).toEqual("122");
+            expect(updatedResult.testTypes[0].certificateNumber).toEqual(null);
+        });
+    });
+
+    context("when inserting a test that is Annual with Certificate, not abandoned, not LEC, not ADR and not Roadworthiness", () => {
```

Comment:
Same applies for when we generate the certificates.
The rules seems to be stating that we generate the certificate when Roadworthiness test is pass